test/functional/lua/hl_spec.lua

```diff
@@ -184,6 +184,10 @@ describe('vim.hl.on_yank', function()
     eq({ win }, api.nvim__ns_get(ns).wins)
     command('wincmd w')
     eq({ win }, api.nvim__ns_get(ns).wins)
+    -- Use a new vim.hl.range() call to cancel the previous timer
```

Comment:
@lewis6991 (just for reference) this sounds like the "ref cleanup" I vaguely worried about in https://github.com/neovim/neovim/pull/32244